low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,500 @@
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import os
+import sys
+import glob
+import time
+import json
+import errno
+import shlex
+import shutil
+import psutil
+import tarfile
+import logging
+import argparse
+import subprocess
+
+from datetime import datetime
+
+from cortx.utils.conf_store import Conf
+from cortx.utils.kv_store import KvStoreFactory
+from framework.base import sspl_constants as sspl_const
+from framework.utils.conf_utils import (
+    Conf, SSPL_CONF, SYSTEM_INFORMATION, SSPL_STATE)
+from framework.utils.ipmi_client import IpmiFactory
+from cortx.utils.process import SimpleProcess
+
+
+# Load bundle request status tracker
+store_type = "json"
+try:
+    bundle_tracker = "/var/cortx/resources/"
+    os.makedirs(bundle_tracker, exist_ok=True)
+except PermissionError as err:
+    print("Failed to create default store directory. %s" % err)
+requests_url = "%s://%s" % (store_type, os.path.join(bundle_tracker,"bundle_requests.json"))
+req_register = KvStoreFactory.get_instance(requests_url)
+req_register.load()
+
+
+class SupportBundleError(Exception):
+    """Generic Exception with error code and output."""
+
+    def __init__(self, rc, message, *args):
+        """Initialize with custom error message and return code."""
+        self._rc = rc
+        self._desc = message % (args)
+
+    def __str__(self):
+        """Format error string."""
+        if self._rc == 0:
+            return self._desc
+        return "SupportBundleError(%d): %s" % (self._rc, self._desc)
+
+
+class BundleRequestHandler:
+    """Handles SSPL support bundle requests."""
+    INPROGRESS = "In-progress"
+    SUCCESS = "Success"
+    FAILED = "Failed"
+
+    def get_processing_status(req_id):
+        """
+        Returns "in-progress" if any request is being processed.
+        Otherwise returns "Success" or "Failed (with reason)" status.
+        """
+        if not req_id:
+            raise SupportBundleError(errno.EINVAL, "Invalid request ID.")
+        status_list = req_register.get(["%s>status" % req_id])
+        status = status_list[0] if status_list else None
+
+        if not status:
+            raise SupportBundleError(
+                errno.EINVAL, "Request ID '%s' not found." % req_id)
+        else:
+            # Set failed status to stale request ID
+            expiry_sec = 1800
```

Comment:
expiry needs to be configurable

Reply:
I have moved it to SSPL constants as it will fairly remain constant. Please suggest if you think it should be moved to somewhere else.